tests/utils/test_augmentation.py

```diff
@@ -0,0 +1,251 @@
+"""Tests for augmentation utilities."""
+
+
+import pandas as pd
+import pytest
+from pytest import param
+
+from baybe.utils.augmentation import (
+    df_apply_dependency_augmentation,
+    df_apply_permutation_augmentation,
+)
+
+
+@pytest.mark.parametrize(
+    ("data", "col_groups", "data_expected"),
+    [
+        param(  # 2 invariant cols and 1 unaffected col
+            {
+                "A": [1, 1],
+                "B": [2, 2],
+                "C": ["x", "y"],
+            },
+            [["A"], ["B"]],
+            {
+                "A": [1, 2, 1, 2],
+                "B": [2, 1, 2, 1],
+                "C": ["x", "x", "y", "y"],
+            },
+            id="2inv_1add",
+        ),
+        param(  # 2 invariant cols with identical values
+            {
+                "A": [1, 1],
+                "B": [2, 2],
+            },
+            [["A"], ["B"]],
+            {
+                "A": [1, 1, 2],
```

Comment:
uff good catch

the utilities do not do any deduplciation and if a row is duplciate also the augmentated rows will be duplciate,

turns out this test was wrong but the equality via merging is a bad idea when expecting duplciates... so i removed it and use a stricter `df.equals`. This required to add the explicit indices in the test configuration, which I did (means the tests now also test the correct index-assignment which is a feature of the utils previously untested)

Reply:
Ok, sounds good! By the way, just FYI in case you are not aware of it: there is also the `pandas.testing` subpackage as part of `pandas` that comes with a lot of dataframe comparison operators. Perhaps there's something in there to help you – if not here, then maybe also in the future